File: 0009-palindrome-number/0009-palindrome-number.kt
class Solution {
    fun isPalindrome(x: Int): Boolean {
        if (x < 0) return false
        if (x == 0) return true
        val lst = ArrayList<Int>()
        var n = x
        while (n > 0) {
            lst.add(n % 10)
            n /= 10
        }
        for (i in 0..lst.lastIndex/2) {
            if (lst[i] != lst[lst.lastIndex-i]) return false
        }
        return true
    }
}

File: 0009-palindrome-number/0009-palindrome-number.rs
impl Solution {
    pub fn is_palindrome(x: i32) -> bool {
        if x < 0 { return false }
        if x == 0 { return true}
        let mut v = Vec::new();
        let mut n = x;
        while n > 0 {
            v.push(n % 10);
            n /= 10;
        }
        let len = v.len();
        
        for i in 0..len/2 {
            if v[i] != v[len-i-1] {
                return false
            }
        }
        return true
    }
}